### ISSUE TITLE
Dropdown : 'Fruits' and same for 'Vegetables' not associated with the items under the combo box.

### ISSUE THREAD

**Issue by 999-Dev** [OfficeDev/office-ui-fabric-react#10921]
**User Experience:**
Screen reader users will face difficulty if labels are not associated with the items under combo box control.

**Test Environment:**
OS: Version: RS5_Prerelease(1903 OS Build 19005.1)
URL: https://developer.microsoft.com/en-us/fabric#/controls/web/dropdown
Edge Browser Version:
Microsoft Edge: 44.19005.1.0
Microsoft EdgeHTML: 18.19005
Chrome Browser Version:
Chrome: 77.0.3865.120

**Usages:**
All Usages

**Repro Steps:**
1.https://developer.microsoft.com/en-us/fabric#/controls/web/dropdown in Edge and turn on Narrator.
2.Navigate to 'Basic input' button and activate it.
3.Navigate to 'Dropdown' link and activate it.
4.'Dropdown' pane will be appeared.
5.Navigate to 'Basic uncontrolled example' combo box control and activate it.
6.Navigate by using up and down arrow keys and verify the issue.

**Actual Result:**
Screen reader does not convey any information about 'Fruits' and 'Vegetables' labels.

**Expected Result:**
'Apple', 'Banana' are not associated with the 'Fruits' and same for 'Vegetables'

**MAS Reference:**
https://microsoft.sharepoint.com/:w:/r/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={4b95a70d-258c-474a-820e-5c8b79d5c0bf}

**Attachment:**
![image](https://user-images.githubusercontent.com/55537167/67203698-67927a00-f429-11e9-8866-62c9f8e619f7.png)
['Fruits' and same for 'Vegetables' are not associated with items.zip](https://github.com/OfficeDev/office-ui-fabric-react/files/3750696/Fruits.and.same.for.Vegetables.are.not.associated.with.items.zip)

**Comment by jdhuntington** [comment]
Hi @999-Dev , thanks for the feedback. I agree that narrator does not mention the relation of "Apple" to "Fruits" in this example. I also checked the native controls with `select`, `option`, and `optgroup` and found narrator's behavior to be nearly identical.

What would you like to hear from narrator in this example?

**Comment by 999-Dev** [comment]
Hi @jdhuntington,
I just want narrator should convey the information about the 'Fruits', So that user can easily understand that we are navigating the 'Fruits'. Same as for 'Vegetables'.

**Comment by micahgodbolt** [comment]
@joschect what's the design need here? Do you feel we still need a bit of clarity on what narrator will say?

**Comment by joschect** [comment]
@micahgodbolt I think so. My question is mainly how groups are expected to interact with the overall number of items in the list.

If I have 10 total items, 5 fruits and 5 vegetables, should selecting the first fruit say "1 of 10" or "1 of 5".

Additionally, what is the correct dom structure? We could group them using a role="group" on the root and have each item in there. Otoh we could keep the same dom structure we have right now and add something like described by so each item knows what header it's described by. 

The easiest and quickest fix is to assume that we can still say "1 of 10", keep the same dom structure and then open up props for label or described-by and then it's up to the consuming component to supply the correct id's and described-by's for all the items.

**Comment by micahgodbolt** [comment]
quote from #11265 dup

> One approach is to have the group (containing div) that contains the header also have an aria-labelledby that points to that header. Not sure if the menu items are structured that way today or if it would need to be reworked to have that structure (I believe it's the latter)

**Comment by micahgodbolt** [comment]
the above approach seems reasonable. it'd only require a bit more work in [`_onRenderLists`](https://github.com/OfficeDev/office-ui-fabric-react/blob/bf72287bf9c3fd69fe791e0ad62b86d638c7a811/packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.base.tsx#L539-L543). The interface would stay the same, but we'd break the return up into divs with headers and items below, making the assumption that everything between headers is in the same group? Or do we split at dividers?

```
header1
item1
item2
divider
item3
header2
item4
item5
````
Would item 3 be inside group 1 or separate? @jspurlin  thoughts?

**Comment by jspurlin** [comment]
> the above approach seems reasonable. it'd only require a bit more work in [`_onRenderLists`](https://github.com/OfficeDev/office-ui-fabric-react/blob/bf72287bf9c3fd69fe791e0ad62b86d638c7a811/packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.base.tsx#L539-L543). The interface would stay the same, but we'd break the return up into divs with headers and items below, making the assumption that everything between headers is in the same group? Or do we split at dividers?
> 
> ```
> header1
> item1
> item2
> divider
> item3
> header2
> item4
> item5
> ```
> 
> Would item 3 be inside group 1 or separate? @jspurlin thoughts?

Good question... My gut reaction is that it should be separate, but the argument could be made that the divider could be in context of the section (e.g. a non-labelled subsection)... I think either approach wouldn't be "wrong". @cschlechty may have some additional thoughts

**Comment by micahgodbolt** [comment]
from a narrator perspective i don't know if it'll change anything. the header is only read off when you first enter the group. regardless of item3 being in header1 or out, narrator output wouldn't change (afaik)

**Comment by jspurlin** [comment]
It would if the group ended when the divider was present, like:
```
<group1start>
<header1>
<item1>
<item2>
<group1end>
<divider>
<item3>
<group2start>
<header2>
<item4>
<item5>
<group2end>
```

VS

```
<group1start>
<header1>
<item1>
<item2>
<divider>
<item3>
<group1end>
<group2start>
<header2>
<item4>
<item5>
<group2end>
```

I think the first approach conveys more of the visible UI to the end user since it will tell them that they are leaving the group (as opposed to them not knowing that divider is present without them having to use virtual navigation to search for (and potentially find) the divider)
